modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/leases/Lease.java

```diff
@@ -153,10 +151,10 @@ public Lease acceptLease(HybridTimestamp to) {
      *
      * @return Denied lease.
      */
-    public Lease denyLease(String proposedCandidate) {
-        assert accepted : "The lease is not accepted: " + this;
+    public Lease denyLease(@Nullable String proposedCandidate) {
+        HybridTimestamp newExpirationTime = accepted ? expirationTime : new HybridTimestamp(System.currentTimeMillis(), 0);
 
-        return new Lease(leaseholder, leaseholderId, startTime, expirationTime, false, true, proposedCandidate, replicationGroupId);
+        return new Lease(leaseholder, leaseholderId, startTime, newExpirationTime, false, accepted, proposedCandidate, replicationGroupId);
```

Comment:
The accepted flag should be set to true.

Reply:
We can't do this: replica must accept the lease and send LeaseGrantMessageResponse before that. Otherwise we can grant the accepted lease to replica that could have not caught up yet.